File: Backend/Models/Goal.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace Backend.Models
{
    public class Goal
    {
        [Key]
        public Guid Id { get; set; }
        public string Name { get; set; } = null!;
        public string Description { get; set; } = null!;
        public DateTime CreatedDate { get; set; }
        public DateTime DueDate { get; set; }
        public bool IsComplete { get; set; }

        [Range(0, 100)]
        public double ProgressPercentage { get; set; } = 0d;

        [Required]
        public List<GoalTask> GoalTasks { get; set; } = null!;

        public List<Tag> Tags { get; set; } = null!;
    }
}

File: Frontend/ProductivNG/src/app/Models/Goal.ts
export class Goal {
  id: number;
  name: string;
  description: string;
  dueDate: Date;
  isComplete: boolean;
  progressPercentage: number;
  constructor(
    id: number,
    name: string,
    description: string,
    deadline: Date,
    completed: boolean,
    progressPercentage: number
  ) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.dueDate = deadline;
    this.isComplete = completed;
    this.progressPercentage = progressPercentage;
  }
}
